scripts/python/utility/get_iabp_from_web.py

```diff
@@ -0,0 +1,260 @@
+    data = ""
+    if index_all == -1:
+        return -1, data
+  
+    #is this the weird (bad) .dat.dat situtation?
+    teststr = data_all[index_all:index_all+8]
+    if teststr == ".dat.dat":
+        indexend = data_all.find('.dat.dat<', index_all+1)
+        if indexend == -1:
+            print("Unexpected lack of .dat.dat<")
+            return -1, data
+        data = data_all[index_all+10:indexend+8]
+    else:
+        indexend = data_all.find('.dat<', index_all+1)
+        if indexend == -1:
+            print("UNexpected lack of .dat<")
+            return -1, data
+        data = data_all[index_all+6:indexend+4]
+    return indexend+10, data
+
+#----------------------------------------------
+def getStation(sfile):
+    cmd = "wget  https://iabp.apl.uw.edu/WebData/" + sfile
+    print(cmd)
+    doCmd(cmd, True)
+
+    # parse contents (not used for anything just yet)
+    with open(sfile, 'r') as file:
+        data_all = file.read()
+        file.close()
+        lines = data_all.splitlines()
+
+        # first line is a header, remaining lines are a time series
+        sh = StationHeader(lines[0], sfile)
+        if sh._ok:
+            lines = lines[1:]
+            st = StationTimeSeries(sh)
+            for l in lines:
+                st.add(l, sfile)
+
+            
+#----------------------------------------------
+def run2(output_path):
+
+    cwd = os.getcwd()
+
+    if (output_path[0:2] != "./" and output_path[0] != '/'):
+        outpath = "./" + output_path
+    else:
+        outpath = output_path
+    makeOrScrub(outpath, True)
+    os.chdir(outpath)
+
+    cmd = "wget  https://iabp.apl.uw.edu/WebData"
+    print(cmd)
+    s = doCmd(cmd, True)
+    if not s:
+      status = False
+
+    stationfiles = []
+    with open("WebData", 'r') as file:
+        data_all = file.read().replace('\n', '')
+    file.close()
+    index_all = 0
+    while index_all < len(data_all):
+        index_all, data = nextStation(data_all, index_all)
+        if (index_all == -1):
+          break;
+        stationfiles.append(data)
+
+    print("Parsed out ", len(stationfiles), " individual station files")
+
+    # pull down all the station files
+    for i in range(len(stationfiles)):
+        getStation(stationfiles[i])
+        
+    print("created ", len(stationfiles), " station files in ", outpath)
+    os.chdir(cwd)
+
+#----------------------------------------------
+def run():
+    # this does not work, but keeping it in for now as a reference
+    #——first open and read “ArcticTable_Current.txt” to get the table of updating buoys:
+    fid=urllib.request.urlopen('https://iabp.apl.uw.edu/TABLES/ArcticTable_Current.txt')
+    table=fid.read()
+    fid.close()
+    s = str(table)
+    tables=s.split('\n')  #The "\n" is the "new line" character. Splitting the table data on it breaks the table into a list of rows.
+    print(tables)
+    return True
+
+    #—— next cycle through the rows in table and download the associated data:
```

Comment:
@davidalbo, [this run](https://github.com/dtcenter/MET/actions/runs/8622097184/job/23632467977?pr=2846) of the newly added SonarQube scanning workflow classified lines 233 - 241 of this file as introducing a new "Bug". They are unreachable since they come after the `return True` statement on line 231.

Here's the [details in the SonarQube Scan](https://needham.rap.ucar.edu/project/issues?branch=refs%2Fpull%2F2846%2Fmerge&resolved=false&inNewCodePeriod=true&types=BUG&open=AY6-eVVg24y5hV7HT_8L&id=MET-GHA). Just log into SonarQube using your GitHub credentials.

Is there any harm in removing lines 233 - 241 to avoid increasing the number of "bugs" in the MET code from 7 to 8?

Note that SonarQube counts up [92 additional code smells](https://needham.rap.ucar.edu/project/issues?branch=refs%2Fpull%2F2846%2Fmerge&inNewCodePeriod=true&resolved=false&types=CODE_SMELL&id=MET-GHA) in these changes as well. But I'm less concerned about code smells than bugs.

Reply:
@JohnHalleyGotway removed this unused code on 4/10/24.